troubleshooting/tutorials/valgrind/README.md

```diff
@@ -0,0 +1,118 @@
+# Tutorial: Fluent Bit Test in AWS EC2 Instance with Valgrind for Memory Leak
+
+## Purpose
+
+Valgrind is a multipurpose code profiling and memory debugging software that runs on Linux. It mainly includes tools such as Memcheck, Callgrind, Cachegrind, etc.
+
+Each tool can complete a task of debugging, detection or analysis. Can detect memory leaks, thread violations and Cache usage. Valgrind debugs programs based on emulation. It obtains control of the actual processor before the application, emulates a virtual processor on the basis of the actual processor, and makes the application run on the virtual processor, thereby Monitor the operation of the application. The application does not know whether the processor is virtual or real, and the application that has been compiled into binary code does not need to be recompiled. Valgrind directly interprets the binary code so that the application runs on it, so that it can check the possible occurrences of memory operations. mistake. So programs running under Valgrind run much slower and use much more memory - for example, programs under the Memcheck tool are more than twice as fast as normal. Therefore, it is best to use Valgrind on a machine with good performance.
+
+Here is the official website for [Valgrind](https://valgrind.org/).
+
+## How to Use Valgrind for Fluent-bit
+
+### Preparation
+
+Install your fluent-bit in AWS EC2 Linux 2 instance, that is, in a linux environment, which also meets the conditions for using valgrind.
+
+### Step for Using Valgrind
```

Comment:
Please note that Fluent Bit has support for valgrind that makes it easier for valgrind to track the stack when Fluent bit switches from one coroutine to another. 

You must compile it with `cmake -DFLB_VALGRIND=On` to get this support. Please ntoe this.